dev-support/create-release/release-build.sh

```diff
@@ -20,114 +20,180 @@
-    rm -rf "${tmp_repo}" &> /dev/null || true
+  if [[ -z "${REPO}" ]] && [[ -n "${MAVEN_LOCAL_REPO}" ]]; then
+    echo "Cleaning up temp repo in '${MAVEN_LOCAL_REPO}'. Set REPO to reuse downloads." >&2
+    rm -f "${MAVEN_SETTINGS_FILE}" &> /dev/null || true
+    rm -rf "${MAVEN_LOCAL_REPO}" &> /dev/null || true
   fi
 }
 
-if [ $# -eq 0 ]; then
+if [ $# -ne 1 ]; then
   exit_with_usage
 fi
 
-if [[ $@ == *"help"* ]]; then
+if [[ "$*" == *"help"* ]]; then
   exit_with_usage
 fi
 
-# Read in the ASF password.
-if [[ -z "$ASF_PASSWORD" ]]; then
-  echo 'The environment variable ASF_PASSWORD is not set. Enter the password.'
-  echo
-  stty -echo && printf "ASF password: " && read ASF_PASSWORD && printf '\n' && stty echo
-fi
+init_locale
+init_java
+init_mvn
+init_python
+# Print out subset of perl version (used in git hooks)
+perl --version | grep 'This is'
 
-# Read in the GPG passphrase
-if [[ -z "$GPG_PASSPHRASE" ]]; then
-  echo 'The environment variable GPG_PASSPHRASE is not set. Enter the passphrase to'
-  echo 'unlock the GPG signing key that will be used to sign the release!'
-  echo
-  stty -echo && printf "GPG passphrase: " && read GPG_PASSPHRASE && printf '\n' && stty echo
-  export GPG_PASSPHRASE
-  export GPG_TTY=$(tty)
-fi
+rm -rf "${PROJECT}"
+
+if [[ "$1" == "tag" ]]; then
+  # for 'tag' stage
+  set -o pipefail
+  set -x  # detailed logging during action
+  check_get_passwords ASF_PASSWORD
+  check_needed_vars PROJECT ASF_USERNAME ASF_PASSWORD RELEASE_VERSION RELEASE_TAG NEXT_VERSION \
+      GIT_EMAIL GIT_NAME GIT_BRANCH
+  ASF_REPO="gitbox.apache.org/repos/asf/${PROJECT}.git"
+  encoded_username="$(python -c "import urllib; print urllib.quote('''$ASF_USERNAME''')")"
+  encoded_password="$(python -c "import urllib; print urllib.quote('''$ASF_PASSWORD''')")"
+  git clone "https://$encoded_username:$encoded_password@$ASF_REPO" -b "$GIT_BRANCH" "${PROJECT}"
+
+  # 'update_releasenotes' searches the project's Jira for issues where 'Fix Version' matches specified
+  # $jira_fix_version. For most projects this is same as ${RELEASE_VERSION}. However, all the 'hbase-*'
+  # projects share the same HBASE jira name.  To make this work, by convention, the HBASE jira "Fix Version"
+  # field values have the sub-project name pre-pended, as in "hbase-operator-tools-1.0.0".
+  # So, here we prepend the project name to the version, but only for the hbase sub-projects.
+  jira_fix_version="${RELEASE_VERSION}"
+  shopt -s nocasematch
+  if [[ "${PROJECT}" =~ ^hbase- ]]; then
+    jira_fix_version="${PROJECT}-${RELEASE_VERSION}"
+  fi
+  shopt -u nocasematch
+  update_releasenotes "$(pwd)/${PROJECT}" "${jira_fix_version}"
+
+  cd "${PROJECT}"
+
+  git config user.name "$GIT_NAME"
+  git config user.email "$GIT_EMAIL"
 
-for env in ASF_USERNAME GPG_PASSPHRASE GPG_KEY; do
-  if [ -z "${!env}" ]; then
-    echo "ERROR: $env must be set to run this script"
-    exit_with_usage
+  # Create release version
```

Comment:
Release or release candidate?

Reply:
This whole block of code, starting with `if [[ "$1" == "tag" ]]; then` is just cut in from what was formerly `release-tag.sh`. Unfortunately, `diff` interleaved it with random chunks of deleted material from `release-build.sh`, so the flow isn't as obvious as one would wish. The answer to your question is item #3 below:

The flow of `tag` is, briefly:
1. Checkout the code
1. Update Release Notes -- use yetus scripts to scan Jira data and update CHANGES.md and RELEASENOTES.md
1. Set maven version in pom.xml for the release or release candidate you are making. This just uses the version you've specified in do-release.sh, which is usually a release candidate version, but it's up to you.
1. Do `git commit`
1. Do `git tag` to make the tag for the release.
1. Set maven version again, this time to the dev version to be used on this branch _after_ the release.
1. Finally, do `git commit` again.

Reply:
k